# Assembling a bacterial genome sequence - exploiting long reads

Previously, we assembled a bacterial genome sequence from about a million pairs of short sequence reads.
This produced a draft-quality genome sequence that was, unfortunately, fragmented into hundreds of contigs.

In the following exercise, we will experience the value of using long sequencing reads to improve the genome assembly.


### Hands on: copy the Galaxy history to create a new Galaxy history


![Copy the Galaxy history 1](<assembly/Screenshot 2025-10-02 at 09.56.25.png>)


See image: [Copy the Galaxy history](<assembly/Screenshot 2025-10-02 at 09.56.25.png>).


![Copy the Galaxy history 2](<assembly/Screenshot 2025-10-02 at 09.57.21.png>)


See image: [Copy the Galaxy history 2](<assembly/Screenshot 2025-10-02 at 09.57.21.png>).


![New Galaxy history](<assembly/Screenshot 2025-10-02 at 09.58.23.png>)

See image: [New Galaxy history](<assembly/Screenshot 2025-10-02 at 09.58.23.png>).


### Hands on: get the long sequence reads

The long Oxford Nanopore sequence reads for this bacterial genome are in the Sequence Read Archive (SRA)
under accession number SRR15305418.


![Getting the sequence reads from SRA into Galaxy](<assembly/Screenshot 2025-10-02 at 10.02.07.png>)

See image: [Getting the sequence reads from SRA into Galaxy](<assembly/Screenshot 2025-10-02 at 10.02.07.png>).







